src/lldp_syncd/daemon.py

```diff
@@ -109,35 +111,84 @@ class ChassisIdSubtypeMap(int, Enum):
         # chassis = int(LldpChassisIdSubtype.chassisComponent) # (unsupported by lldpd)
         local = int(LldpPortIdSubtype.local)
 
+    def get_sys_capability_list(self, if_attributes):
+        """
+        Get a list of capabilities from interface attributes dictionary.
+        :param if_attributes: interface attributes
+        :return: list of capabilities
+        """
+        try:
+            # [{'enabled': ..., 'type': 'capability1'}, {'enabled': ..., 'type': 'capability2'}]
+            capability_list = if_attributes['chassis'].values()[0]['capability']
+            # {'enabled': ..., 'type': 'capability'}
+            if not isinstance(capability_list, list):
+                capability_list = [capability_list]
+        except KeyError:
+            logger.error("Failed to get system capabilities")
+            return None
+        return capability_list
+
+    def parse_sys_capabilities(self, capability_list, enabled=False):
+        """
+        Get a bit map of capabilities, accoding to textual convention.
+        :param capability_list: list of capabilities
+        :param enabled: if true, consider only the enabled capabilities
+        :return: string representing a bit map
+        """
+        # if chassis is incomplete, missing capabilities
+        # TODO check if it does not break snmpagent
+        if not capability_list:
+            return None
+
+        sys_cap = 0x00
+        for capability in capability_list:
+            try:
+                if (not enabled) or capability["enabled"]:
+                    sys_cap |= 128 >> LldpSystemCapabilitiesMap[capability["type"].lower()]
+            except KeyError:
+                logger.warning("Unknown capability {}".format(capability["type"]))
+        return "%0.2X 00" % sys_cap
+
     def __init__(self, update_interval=None):
         super(LldpSyncDaemon, self).__init__()
         self._update_interval = update_interval or DEFAULT_UPDATE_INTERVAL
         self.db_connector = SonicV2Connector()
         self.db_connector.connect(self.db_connector.APPL_DB)
 
+        self.chassis_cache = {}
+        self.interfaces_cache = {}
+
     def source_update(self):
         """
         Invoke lldpctl and format as JSON
         """
         cmd = ['/usr/sbin/lldpctl', '-f', 'json']
         logger.debug("Invoking lldpctl with: {}".format(cmd))
+        cmd_local = ['/usr/sbin/lldpcli', '-f', 'json', 'show', 'chassis']
+        logger.debug("Invoking lldpcli with: {}".format(cmd_local))
 
-        try:
-            # execute the subprocess command
-            lldpctl_output = subprocess.check_output(cmd)
-        except subprocess.CalledProcessError:
-            logger.exception("lldpctl exited with non-zero status")
-            return None
+        def scrap_output(cmd):
```

Comment:
>scrap_output [](start = 12, length = 12)

Could you not use embedded function? suggest a name _scrap_output to indicate it is for internal use.